shared.js

```diff
@@ -108,6 +119,78 @@ var Shared = module.exports = {
 			try {
 				/* If theres an error, show that.
 				   If not, show the type along with the result */
+				if (result.isJSEval) {
+					if (result.reason) {
+						reply = result.reason;
+					}
+					else if (!result.success) {
+						reply = result.text;
+					}
+					else {
+						reply = result.text;
+					}
+
+					if (reply.length > 100) {
+						var visibleReply = reply.slice(0, 80);
+						var formattedCode = code;
+
+						function markdownEscape(str){
+							return str.replace(/```/g, '{triple backticks escaped}');
+						}
+
+						var gistFile = "";
+						gistFile += "For this code:";
+						gistFile += "\n\n";
+						gistFile += "```js\n";
+						gistFile += markdownEscape(formattedCode);
+						gistFile += "\n";
+						gistFile += "```";
+						gistFile += "\n\n";
+						gistFile += "This was the "
+							+ (result.success ? 'output' : 'error')
+							+ ':';
+						gistFile += "\n\n";
+						gistFile += "```";
+						gistFile += "\n";
+						gistFile += markdownEscape(result.text);
+						gistFile += "\n";
+						gistFile += "```";
+						gistFile += "\n\n";
+
+						Gist.createGist(gistFile, {
+							filename: result.success ? "result.md" : "error.md",
+							description: "result of " + context.sender.name + "'s code"
+						})
+						.then(function(url){
+							var fullReply = cleanReply(visibleReply) + " ... " + url;
+							context.channel.send_reply(context.intent, fullReply, {truncate: false});
+						})
+						.catch(function(err){
+							var fullReply = cleanReply(visibleReply) + " ... <unable to create gist url>";
+							context.channel.send_reply(context.intent, fullReply, {truncate: false});
+						});
+					}
+					else {
+						context.channel.send_reply(context.intent, cleanReply(reply), {truncate: true});
+					}
+					return;
+
+					function cleanReply(text){
```

Comment:
Can this be defined outside of the `execute_js` function so that it's not newly created every time?